File: app/src/main/java/com/shmedo/mcloudapp/common/ui/activity/WebViewActivity.java
package com.shmedo.mcloudapp.common.ui.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.webkit.WebView;
import android.widget.TextView;

import androidx.appcompat.widget.Toolbar;

import com.hjq.toast.ToastUtils;
import com.shmedo.mcloudapp.R;

import butterknife.BindView;

public class WebViewActivity extends BaseActivity {
    private static final String ARG_PARAM1 = "param1";

    @BindView(R.id.toolbar)
    Toolbar mToolbar;

    @BindView(R.id.toolbar_title)
    TextView mToolbarTitle;

    @BindView(R.id.webView)
    WebView webView;

    private String url;

    public static void startActivity(Context context, String url) {
        Intent intent = new Intent(context, WebViewActivity.class);
        intent.putExtra(ARG_PARAM1, url);
        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
        context.startActivity(intent);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_webview;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setToolBar(R.id.toolbar);
        initWebView();
        parseIntent();
    }

    private void initWebView() {
        webView.getSettings().setJavaScriptEnabled(true);
    }

    private void parseIntent() {
        Intent intent = getIntent();
        if (intent.getExtras() != null && intent.getExtras().containsKey(ARG_PARAM1)) {
            url = intent.getStringExtra(ARG_PARAM1);
            loadData();
        }
    }

    public void loadData() {
        if (TextUtils.isEmpty(url)) {
            ToastUtils.show("地址错误，地址不能为空");
        } else {
            webView.loadUrl(url);
        }
    }
}

File: app/src/main/java/com/shmedo/mcloudapp/common/activity/WebviewActivity.kt
package com.shmedo.mcloudapp.common.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.LinearLayout
import androidx.activity.viewModels
import com.just.agentweb.AgentWeb
import com.kunminx.architecture.ui.page.DataBindingConfig
import com.shmedo.mcloudapp.BR
import com.shmedo.mcloudapp.R
import com.shmedo.mcloudapp.common.viewmodel.state.EmptyViewModel
import com.shmedo.mcloudapp.databinding.ActivityWebviewBinding

class WebviewActivity : BaseActivity() {
    private val binding: ActivityWebviewBinding by lazy { getBinding() as ActivityWebviewBinding }
    private val mStates: EmptyViewModel by viewModels()

    private lateinit var mAgentWeb: AgentWeb

    override fun getDataBindingConfig(): DataBindingConfig {
        return DataBindingConfig(R.layout.activity_webview, BR.vm, mStates)
    }

    override fun initView(savedInstanceState: Bundle?) {
        setToolBar(binding.llToolbar.toolbar)
        if (intent.extras != null) {
            val title = intent.getStringExtra(ARG_TITLE)
            val url = intent.getStringExtra(ARG_URL)
            binding.llToolbar.toolbar.title = title
            mAgentWeb = AgentWeb.with(this)
                .setAgentWebParent(binding.container, LinearLayout.LayoutParams(-1, -1))
                .useDefaultIndicator()
                .createAgentWeb()
                .ready()
                .go(url)
        }
        binding.llToolbar.toolbar.setNavigationOnClickListener {
            finish()
        }
    }

    override fun onPause() {
        mAgentWeb.webLifeCycle.onPause()
        super.onPause()
    }

    override fun onResume() {
        mAgentWeb.webLifeCycle.onResume()
        super.onResume()
    }

    override fun onDestroy() {
        super.onDestroy()
        mAgentWeb.webLifeCycle.onDestroy()
    }

    override fun onBackPressed() {
        if (mAgentWeb.webCreator.webView.canGoBack()) {
            mAgentWeb.webCreator.webView.goBack()
        } else super.onBackPressed()
    }

    companion object {
        private const val ARG_TITLE = "arg_title"
        private const val ARG_URL = "arg_url"
        fun startActivity(context: Context, title: String, url: String) {
            val intent = Intent(context, WebviewActivity::class.java)
            intent.putExtra(ARG_TITLE, title)
            intent.putExtra(ARG_URL, url)
            intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP)
            context.startActivity(intent)
        }
    }
}